hardware_interface/include/hardware_interface/internal/resource_manager.h

```diff
@@ -80,9 +80,9 @@ class ResourceManager : public ResourceManagerBase
   {
     std::vector<std::string> out;
     out.reserve(resource_map_.size());
-    for(typename ResourceMap::const_iterator it = resource_map_.begin(); it != resource_map_.end(); ++it)
+    for (const auto& pair : resource_map_)
```

Comment:
When `range-based for`-ing a `map` I typically like to give the pair a very verbose name, in this case it could be something like
```suggestion
    for (const auto& name_and_joint_handle : resource_map_)
```

This helps understanding what `.first` and `.second` refers to down the line. The common argument against this is that we pair naming with the semantics of that map but I'd argue that's kind of the whole point of programming languages more advanced than bytecode, we can name things to convey context.

Reply:
Sounds good to me, will update the names to be more meaningful.

Reply:
Done in 94e3092, though I used a slightly different name